compiler/sem/semgnrc.nim

```diff
@@ -324,19 +329,24 @@ proc semGenericStmt(c: PContext, n: PNode,
     # Consider 'when declared(globalsSlot): ThreadVarSetValue(globalsSlot, ...)'
     # in threads.nim: the subtle preprocessing here binds 'globalsSlot' which
     # is not exported and yet the generic 'threadProcWrapper' works correctly.
-    let flags = if mixinContext: flags+{withinMixin} else: flags
-    for i in first..<result.safeLen:
-      result[i] = semGenericStmt(c, result[i], flags, ctx)
+    if not result.isError:
+      captureError c.config, result:
+        let flags = if mixinContext: flags+{withinMixin} else: flags
+        for i in first..<result.safeLen:
```

Comment:
Personally, I think the real pattern might be transform children (or a range), with some map proc/block, and on error wrap the whole thing.

It's a combination of flatmap and fold applying a semantic analysis function. I can't even think of a name.

result = foo(result, range, proc)

Could make the first param a var, but that made me feel like a monster.

Reply:
Yeah, just often it's not cleanly over a range, but some custom iteration, hence I chose the template approach.